### PR TITLE
Feature/UUID management + sampling pipeline changes

### PR DESCRIPTION
**Pull request by djtfmartin**
UUID management and sampling pipeline changes to avoid using MapDB

### CODE REVIEW

File: pipelines/src/main/java/au/org/ala/kvs/cache/ALACollectionKVStoreFactory.java [hunk 1]
```diff
@@ -49,8 +49,8 @@ public ALACollectionMatch get(ALACollectionLookup key) {
                         return service.lookupCodes(key.getInstitutionCode(), key.getCollectionCode());
                     } catch (Exception ex) {
                         //this is can happen for bad data and this service is suspectible to http 404 due to the fact
-                        // it takes URL parameters from the raw data. So log and carry on for now.we
-                        LOG.error("Error contacting the collectory service with insitutionCode {} and collection code {} Message: {}",
+                        // it takes URL parameters from the raw data. So log and carry on for now.
+                        LOG.error("Error contacting the collectory service with institutionCode {} and collectionCode {} Message: {}",
```
Comment (qifeng-bai) [inline pipelines/src/main/java/au/org/ala/kvs/cache/ALACollectionKVStoreFactory.java]: LOG or log?

File: pipelines/src/main/java/au/org/ala/pipelines/beam/ALAUUIDMintingPipeline.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+                        .to(alaRecordDirectory + "_new/interpret")
+                        .withSuffix(".avro")
+                        .withCodec(BASE_CODEC));
+
+        log.info("Running the pipeline");
+        PipelineResult result = p.run();
+        result.waitUntilFinish();
+        MetricsHandler.saveCountersToTargetPathFile(options, result.metrics());
+
+        //TODO duplicate check of ALAUUIDRecord entries and report them.....
```
Comment (ansell) [inline pipelines/src/main/java/au/org/ala/pipelines/beam/ALAUUIDMintingPipeline.java:155]: If this is being skipped in this PR, there should be an issue link here to make sure that we can track it being implemented in the future.

File: pipelines/src/main/java/au/org/ala/pipelines/beam/ALAUUIDMintingPipeline.java [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        }
+    }
+
+    /**
+     * Match the darwin core term which has been supplied in simple camel case format e.g. catalogNumber.
+     *
+     * @param name
+     * @return
+     */
+    static Optional<DwcTerm> getDwcTerm(String name){
```
Comment (ansell) [inline pipelines/src/main/java/au/org/ala/pipelines/beam/ALAUUIDMintingPipeline.java:240]: This method needs to be removed, and the terms need to be matched directly, not only if they are recognised Darwin Core Terms.

CC @peggynewman @RobinaSanderson @sadeghim

File: pipelines/src/main/java/au/org/ala/pipelines/beam/ALAVerbatimToInterpretedPipeline.java [hunk 4]
```diff
@@ -148,13 +145,13 @@ public static void run(InterpretationPipelineOptions options) {
             .apply("Check verbatim transform condition", metadataTransform.checkMetadata(types))
             .apply("Convert into view", View.asSingleton());
 
+    // Interpret and write all record types
     PCollection<ExtendedRecord> uniqueRecords = metadataTransform.metadataOnly(types) ?
         verbatimTransform.emptyCollection(p) :
         p.apply("Read ExtendedRecords", verbatimTransform.read(options.getInputPath()))
             .apply("Read occurrences from extension", OccurrenceExtensionTransform.create())
             .apply("Filter duplicates", UniqueIdTransform.create());
```
Comment (ansell) [inline pipelines/src/main/java/au/org/ala/pipelines/beam/ALAVerbatimToInterpretedPipeline.java:153]: How is `UniqueIdTransform` related to UUIDs?

**Pull request merged**